src/app/pages/psbt-request/components/psbt-decoded-request/psbt-decoded-request-views/match-inputs-and-outputs.ts

```diff
@@ -0,0 +1,92 @@
+import * as btc from '@scure/btc-signer';
+
+import { isDefined } from '@shared/utils';
+
+import { PsbtInputForUi } from './use-psbt-decoded-request';
+
+export interface InputOutputPair {
+  inputs: PsbtInputForUi[];
+  output: btc.TransactionOutputRequired;
+}
+
+function consolidateInputsWithSameAddress(inputs: PsbtInputForUi[]): PsbtInputForUi[] {
+  const utxo: Record<string, PsbtInputForUi> = {};
+  inputs.forEach(({ address, value, ...rest }) => {
+    if (isDefined(utxo[address])) {
+      utxo[address] = {
+        ...utxo[address],
+        value: utxo[address].value + value,
+      };
+    } else {
+      utxo[address] = { address, value, ...rest };
+    }
+  });
+  return Object.values(utxo);
+}
+
+interface MatchInputsOutputsArgs {
+  psbtInputs: PsbtInputForUi[];
+  unsignedOutputs: btc.TransactionOutputRequired[];
+}
+export function matchInputsOutputs({ psbtInputs, unsignedOutputs }: MatchInputsOutputsArgs): {
+  fee: number;
+  inputOutputPairs: InputOutputPair[];
+} {
```

Comment:
This is fairly complex so would encourage unit tests.

Separately, I wonder if some of this logic might better belong elsewhere. It's nested quite deep in the component tree. Maybe we could separate out higher in some general `psbt` logic folder?

Reply:
It is always `10000` for the sell or buy. The inscription utxo itself may not be 10000. But the new input/output is.

Reply:
> This is fairly complex so would encourage unit tests.
> 
> Separately, I wonder if some of this logic might better belong elsewhere. It's nested quite deep in the component tree. Maybe we could separate out higher in some general `psbt` logic folder?

Do you anticipate psbts being used elsewhere in the code? Not sure why I would move unless it will be used elsewhere?

Reply:
Not suggesting moving away from the feature, but from the deeply tested view, something like `src/pages/psbt-request/utils`?

Not necessary, just promoting decoupling views/business logic.

Reply:
Interesting, ok, I thought better to locate near where it is being used if not shared.

Reply:
I will move it out ...I first thought you meant out into shared utils.

Reply:
Made a `hooks` folder higher up, think that is best bc used there and can locate hooks together.